### PR TITLE
Fix #6787

### LINKED ISSUE #6787: [Bug]: Starting with v1.52.0 logging with Langfuse leads to error: "team_id missing from team"
### What happened?

First of all thanks for the great work. Unfortunately I encounter a small bug starting with v1.52.0

I need to log to Langfuse and for this I use the following config:

litellm_settings: 
  default_team_settings: 
    - team_id: c91e32bb-0f2a-4aa1-86c4-307ca2e03ea3 
      success_callback: ["langfuse"]
      failure_callback: ["langfuse"]
      langfuse_public_key: ***
      langfuse_secret: ***
      
This then leads to the following error (see also added Stacktrace):
Exception: team_id missing from team

Do you have any ideas why this happens? Thanks in advance and cheers

### Relevant log output

```shell
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 3295, in chat_completion
    data = await add_litellm_data_to_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/litellm_pre_call_utils.py", line 512, in add_litellm_data_to_request
    team_config = await proxy_config.load_team_config(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 1472, in load_team_config
    raise Exception(f"team_id missing from team: {team}")
Exception: team_id missing from team: {'success_callback': ['langfuse'], 'failure_callback': ['langfuse'], 'langfuse_public_key': 'pk-lf-008c75dc-f7a6-4e8b-8f3a-3fa8631cedf6', 'langfuse_secret': 'sk-lf-64ca0c4b-8a4f-4c81-896f-7e59864d057c'}
07:38:24 - LiteLLM Proxy:ERROR: _common.py:120 - Giving up chat_completion(...) after 1 tries (litellm.proxy._types.ProxyException)
INFO:     10.13.0.29:36316 - "POST /v1/chat/completions HTTP/1.1" 500 Internal Server Error
07:38:24 - LiteLLM Proxy:ERROR: proxy_server.py:3482 - litellm.proxy.proxy_server.chat_completion(): Exception occured - team_id missing from team: {'success_callback': ['langfuse'], 'failure_callback': ['langfuse'], 'langfuse_public_key': 'pk-lf-008c75dc-f7a6-4e8b-8f3a-3fa8631cedf6', 'langfuse_secret': 'sk-lf-64ca0c4b-8a4f-4c81-896f-7e59864d057c'}
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 3295, in chat_completion
    data = await add_litellm_data_to_request(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/litellm_pre_call_utils.py", line 512, in add_litellm_data_to_request
    team_config = await proxy_config.load_team_config(
                  ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/litellm/proxy/proxy_server.py", line 1472, in load_team_config
    raise Exception(f"team_id missing from team: {team}")
Exception: team_id missing from team
```

### Twitter / LinkedIn details

_No response_

### PR DESCRIPTION
**Pull request by Manouchehri**
_Seems_ to fix #6787 for me.

### CODE REVIEW

**Comment by themrzmaster** [comment]
any update on this? @krrishdholakia  
having the same problem

File: litellm/proxy/proxy_server.py [hunk 1]
```diff
@@ -1475,7 +1475,7 @@ async def load_team_config(self, team_id: str):
             if "team_id" not in team:
                 raise Exception(f"team_id missing from team: {team}")
             if team_id == team["team_id"]:
-                team_config = team
+                team_config = copy.deepcopy(team)
```
Comment (ghost) [inline litellm/proxy/proxy_server.py:1478]: hey @Manouchehri sorry just seeing this - do you know what was causing this?

Comment (themrzmaster) [reply litellm/proxy/proxy_server.py:1478]: I guess this solution does not provoke errors but the logging does not work

Comment (Manouchehri) [reply litellm/proxy/proxy_server.py:1478]: I don't know exactly what's causing this. I'm guessing it's a race condition with multiple threads mutating the same dictionary.

Comment (Manouchehri) [reply litellm/proxy/proxy_server.py:1478]: The logging is working for me.

**Comment by ghost** [comment]
@Manouchehri closing this PR as the current fix seems susceptible to regressions (not addressing root cause). Happy to reopen, based on what we discover in #6787. My understanding of the current issue is that it's being caused by an override of the proxy config by the db settings.

**Comment by Manouchehri** [comment]
Okay. It's been several weeks at this point (months iirc since the private report), so I personally felt a hacky fix was better than no fix.

**Pull request closed without merging**